feat: 홈페이지 UI 구현 완료

Yanguk opened the pull request:
## 작업 사항
- 홈페이지 UI 작업을 완료하였습니다.
- react router를 이용한 초기 라우터 셋팅을 하였습니다.
- 라이브러리 추가항목
   - react-router-dom @v6
   - react-icons
   - styled-components

<img width="1552" alt="image" src="https://user-images.githubusercontent.com/99335782/196039826-b333b21c-c1bc-465c-a4d2-8f3160037b2f.png">
<img width="1552" alt="image" src="https://user-images.githubusercontent.com/99335782/196039834-8a4ae91b-c4a3-4fec-b3ee-71d766df467c.png">



## 잘 봐주세요
- 폴더 구조 
- app.js 에서의 라우터 셋팅
- HomePage 컴포넌트

## PR 전 확인사항
[x] 가장 최신 브랜치를 pull했습니다.
[x] base 브랜치명을 확인했습니다.
[x] 코드 컨벤션을 모두 지켰습니다.
[x] 적절한 라벨이 있습니다.
[x] assignee가 있습니다.

## 비고
- 프론트 작업 스타트 시작지점이라 전체적인 폴더 구조 및 라우터 셋팅도 변경사항에 추가되었습니다.
- 그외 주요 작업물은 HomePage 컴포넌트 입니다.

Yanguk committed:
feat: nested routes로 페이지및 레이아웃 셋팅

package.json 변경: react-router-dom @v6 추가되었습니다

Yanguk committed:
feat: GlobalStyle 추가 및 styled component 도입

package.json 변경: styled-components 추가

Yanguk committed:
feat: 홈페이지 작업 아직 진행중..

홈페이지 작업중입니다.
작업 공유를 위하여 커밋 내역 남깁니다.

Yanguk committed:
feat: 홈페이지 UI 구현 완료

assets 폴더안에 서비스앱 이미지 추가
package.jso 추가사항: react-icons 라이브러리 추가

Yanguk committed:
refactor: css 순서 컨벤션에 따라 수정

ssong-yi commented on src/common/constants/theme.js:
```diff
@@ -0,0 +1,9 @@
+  white: "#feffff",
+};
+
+export default theme;
```
편하게 쓸 수 있도록 자주 쓰는 컬러를 넣어주셨군요! 👍

ssong-yi commented on src/common/components/Navbar/index.js:
```diff
@@ -0,0 +1,76 @@
+`;
+
+const GoToHome = styled(Link)`
+  display: block;
```
저희 `styled-components` 태그 순서대로 밑에 변수 선언하면 더 보기 좋을 것 같은데, 어떠세요?

예를 들어서,
```javascript
const Container = styled.nav``
const GoToHome = styled.(Link)``
const HomeIcon = styled(FaHeadphones)``
```

Yanguk replied:
넵 좋아요 반영하였습니다

rktnsinger commented on src/app/App.js:
```diff
@@ -0,0 +1,42 @@
+  );
+}
+
+export default App;
```
BE쪽 컨벤션과 맞춘다면, 이렇게 밑에서 export하는게 맥락에 잘 맞는 것 같습니다!! 👍

rktnsinger commented on src/app/App.js:
```diff
@@ -0,0 +1,42 @@
+      </Route>
+
+      <Route path="/error" element={<ErrorPage />} />
+      <Route path="*" element={<NotFoundPage />} />
```
에러 페이지까지 벌써 다 세팅...너무 감사합니다 👍

rktnsinger commented on src/app/App.js:
```diff
@@ -0,0 +1,42 @@
+        <Route index element={<HomePage />} />
+        <Route path="docs" element={<GuidePage />} />
+      </Route>
+
```
혹시 컴포넌트 개행 기준은 어떻게들 하시는지 같이 이야기 나누고싶어요!

Yanguk replied:
논의한데로 개행 없이 수정하겠습니다

rktnsinger commented on src/app/App.js:
```diff
@@ -0,0 +1,42 @@
+import ChannelDetailPage from "../common/pages/ChannelDetailPage";
+import ErrorPage from "../common/pages/ErrorPage";
+import NotFoundPage from "../common/pages/NotFoundPage";
+import GuidePage from "../common/pages/GuidePage";
```
혹시 아래 페이지 라우트 순서와 임포트 순서를 맞추는건 어떨까요? 😄

Yanguk replied:
순서 맞추겠습니다 그게 더 보기 깔끔할 것 같네요

rktnsinger commented on src/common/pages/HomePage/index.js:
```diff
@@ -0,0 +1,132 @@
+import theme from "../../constants/theme";
+
+export default function HomePage() {
+  const handleScrollToDown = () => {
```
저는 개인적으론 handleScrollDown도 충분히 의미가 잘 전달될 것 같아요!

Yanguk replied:
좋습니다~

rktnsinger reviewed (approved):
고생 많으셨습니다..그저 빛..테크리드 양욱님 ⭐️

저희 오전에 논의사항 반영해주시고 merge 해주시면 될 것 같습니다~!

오늘 하루도 화이팅!!!

Yanguk committed:
refactor: 코드리뷰 반영

폴더 구조를 논의한데로 구성 하였습니다.
styled component 순서를 사용한 순서대로 선언순서를 변경하였습니다.
App.js에서 라우트 개행 삭제
body태그 background-color 설정

ssong-yi commented on src/pages/HomePage/index.js:
```diff
@@ -0,0 +1,132 @@
+  );
+}
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  width: 100%;
+  height: calc(200% - 60px);
+  background-color: ${theme.skyBlue};
+`;
```
속성 선언 순서 컨벤션에 따라 잘 작성해주신 것 같습니다! 👍

ssong-yi commented on src/pages/HomePage/index.js:
```diff
@@ -0,0 +1,132 @@
+      <StartGuideWrapper>
+        <TitleWrapper>
+          <Title>실시간 그룹콜</Title>
+          <Title>지금 바로 시작하세요</Title>
```
```javascript
<Title>
  실시간 그룹콜
  <br />
  지금 바로 시작하세요
</Title>
```

이렇게 하면 Title 한 번만 써도 될 것 같아요!

Yanguk replied:
오! 좋네요! 반영하겠습니다

ssong-yi commented on src/layout/HomeLayOut/index.js:
```diff
@@ -0,0 +1,19 @@
+
+import Navbar from "./Navbar";
+
+export default function HomeLayOut() {
+  return (
+    <Container>
+      <Navbar />
+      <Outlet />
+    </Container>
+  );
+}
+
```
이 컴포넌트 덕분에 Outlet 을 라우터 여러개에 공통으로 적용하기 편리해진 것 같습니다!

ssong-yi reviewed (approved):
라우터 확인 완료했습니다!

Yanguk committed:
refactor: 코드리뷰 반영

homepage 컴포넌트 title 에 <br /> 태그 추가
